File: app/models/event.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.db.database import Base

class Event(Base):
    __tablename__ = "events"
    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    title = Column(String(255), nullable=False)
    start_time = Column(DateTime, nullable=False)
    end_time = Column(DateTime, nullable=False)
    location = Column(String(255))
    color_code = Column(String(7))

    owner = relationship("User", back_populates="events")
    notes = relationship("Note", back_populates="event")

File: Mobile/TODO_App/app/src/main/java/com/ysr_klc/todo_app/model/Event.kt
package com.ysr_klc.todo_app.model

import com.google.gson.annotations.SerializedName

data class Event(
    @SerializedName("id") val id: Int,
    @SerializedName("user_id") val userId: Int,
    @SerializedName("title") val title: String,

    @SerializedName("start_time") val startTime: String,
    @SerializedName("end_time") val endTime: String,

    @SerializedName("location") val location: String?,
    @SerializedName("color_code") val colorCode: String?
)


data class CreateUpdateEvent(
    @SerializedName("user_id") val userId: Int,
    @SerializedName("title") val title: String,

    @SerializedName("start_time") val startTime: String,
    @SerializedName("end_time") val endTime: String,

    @SerializedName("location") val location: String?,
    @SerializedName("color_code") val colorCode: String?
)